stl/inc/algorithm

```diff
@@ -1525,7 +1525,7 @@ _NODISCARD _CONSTEXPR20 _FwdIt search_n(
                 _Iter_diff_t<_FwdIt> _Count1 = _Count_diff;
                 auto _UMid                   = _UFirst;
 
-                while (_UOld_first != _UFirst && _Pred(*(_UFirst - 1), _Val)) { // back up over any skipped prefix
+                while (_UOld_first != _UFirst && _Pred(*_Prev_iter(_UFirst), _Val)) { // back up over any skipped prefix

```

Comment:
Is there some reason not to also make this change in the `!_HAS_IF_CONSTEXPR` version of this function? (at line 1609)

Reply:
thanks @mnatsuhara and @CaseyCarter , forgot this one